staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1034,6 +1116,29 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	return ret, nil
 }
 
+func writeStorageVersion(ctx context.Context, crdInfo *crdInfo) {
+	// wait for previous handlers to complete
+	time.Sleep(2 * time.Minute) // make this use an old waitgroup or some such
+
+	// now use a client to write the storage version synchronously.
+	// the context is important because we need to ensure that later
+	for i := 0; i < 10000000; i++ {
+		crdInfo.storageVersionUpdateLock.Lock()
```

Comment:
We will be synchronously handling multiple SV updates of the same CRD.. which might be ok since these are triggered as a part of the CR-request flow meaning there are actual CR requests depending on the intermediate SV update.

I remember previously we had said that we can skip intermediate SV updates and just jump to processing the latest one always, hence the thought about are we ok with doing sync. SV updates here?